src/samples/agents/sample-agent/index.ts

```diff
@@ -0,0 +1,182 @@
+  TaskState,
+  TaskStatusUpdateEvent,
+  Message
+} from "../../../index.js";
+import {
+  InMemoryTaskStore,
+  TaskStore,
+  AgentExecutor,
+  RequestContext,
+  ExecutionEventBus,
+  DefaultRequestHandler
+} from "../../../server/index.js";
+import { A2AExpressApp } from "../../../server/express/index.js";
+
+/**
+ * SampleAgentExecutor implements the agent's core logic.
+ */
+class SampleAgentExecutor implements AgentExecutor {
+  private cancelledTasks = new Set<string>();
+
+  public cancelTask = async (
+    taskId: string,
+    eventBus: ExecutionEventBus,
+  ): Promise<void> => { };
+
+  async execute(
+    requestContext: RequestContext,
+    eventBus: ExecutionEventBus
+  ): Promise<void> {
+    const userMessage = requestContext.userMessage;
+    const existingTask = requestContext.task;
+
+    // Determine IDs for the task and context
+    const taskId = requestContext.taskId;
+    const contextId = requestContext.contextId;
+
+    console.log(
+      `[SampleAgentExecutor] Processing message ${userMessage.messageId} for task ${taskId} (context: ${contextId})`
+    );
+
+    // 1. Publish initial Task event if it's a new task
+    if (!existingTask) {
+      const initialTask: Task = {
+        kind: 'task',
+        id: taskId,
+        contextId: contextId,
+        status: {
+          state: 'submitted',
+          timestamp: new Date().toISOString(),
+        },
+        history: [userMessage], // Start history with the current user message
+        metadata: userMessage.metadata, // Carry over metadata from message if any
+      };
+      eventBus.publish(initialTask);
+    }
+
+    // 2. Publish "working" status update
+    const workingStatusUpdate: TaskStatusUpdateEvent = {
+      kind: 'status-update',
+      taskId: taskId,
+      contextId: contextId,
+      status: {
+        state: 'working',
+        message: {
+          kind: 'message',
+          role: 'agent',
+          messageId: uuidv4(),
+          parts: [{ kind: 'text', text: 'Processing your question' }],
+          taskId: taskId,
+          contextId: contextId,
+        },
+        timestamp: new Date().toISOString(),
+      },
+      final: false,
+    };
+    eventBus.publish(workingStatusUpdate);
+
+    // 3. Publish final task status update
+    const agentReplyText = "The sample agent correctly processed your request.";
+    console.info(`[SampleAgentExecutor] Prompt response: ${agentReplyText}`);
+ 
+    const agentMessage: Message = {
+      kind: 'message',
+      role: 'agent',
+      messageId: uuidv4(),
+      parts: [{ kind: 'text', text: agentReplyText }],
+      taskId: taskId,
+      contextId: contextId,
+    };
+
+    const finalUpdate: TaskStatusUpdateEvent = {
+      kind: 'status-update',
+      taskId: taskId,
+      contextId: contextId,
+      status: {
+        state: 'completed',
+        message: agentMessage,
+        timestamp: new Date().toISOString(),
+      },
+      final: true,
+    };
+    eventBus.publish(finalUpdate);
```

Comment:
Let's have a small artificial delay here same as in [this sample](https://github.com/a2aproject/a2a-tck/blob/50afdd61169b3a6e099df5147d3282b0acf91307/python-sut/tck_core_agent/agent_executor.py#L82).